File: BW/10월2주차/카펫.js
/* 구글링 */
function solution(brown, yellow) {
    var answer = [];
    var total = brown+yellow;
    
    for(y=3; ; y++){
        // 예외처리
        if(total%y != 0) continue;
        var x=total/y;
        if((x-2) * (y-2) === yellow){
            answer[0] = x;
            answer[1] = y;
            break;
        }
    }

    return answer;
}

File: MK/10월 2주차/카펫.java
/**
 * 카펫
 * https://school.programmers.co.kr/learn/courses/30/lessons/42842
 */
class Solution {
    public int[] solution(int brown, int yellow) {
        int start = brown-4;
        int r = start-2;
        int c = 2;

        while (true) {
            int a=(int) r/2;
            int b=(int) c/2;
            if (a*b==yellow) {
                r = a+2;
                c = b+2;
                break;
            }
            r-=2;
            c+=2;
        }
        return new int[] {r,c};
    }
}